macro/src/lib.rs

```diff
@@ -256,36 +292,40 @@ impl FieldParser {
                     segments,
                     leading_colon: type_path.path.leading_colon,
                 };
+                let inner_mmio_path_doc_ref = inner_mmio_path
+                    .to_token_stream()
+                    .to_string()
+                    .replace(" :: ", "::");
                 self.bound_checks.push(quote! {
                     derive_mmio::is_mmio::<#inner_mmio_path>();
                 });
                 let steal_func_name = format_ident!("steal_{}", field_ident);
                 quote! {
-                    #[doc = "Obtain a reference to the inner MMIO field `"]
-                    #[doc = stringify!(#field_ident)]
-                    #[doc = "`."]
-                    #[doc = "# Lifetime"]
+                    #[doc = "Obtain a reference to the inner MMIO field "]
```

Comment:
The field is called `#ident`, but is of type `#inner_mmio_path_doc_ref`, so this wording could be clearer.

Currently I get:

> Obtain a reference to the inner MMIO field \[inner::MmioUartBank\]

I think it would be better to have:

> Obtain a reference to the inner MMIO field `bank_1`, which is of type \[inner::MmioUartBank\]

Reply:
I just updated this to be similar to the other fields.